### ISSUE TITLE
Weird Results After Training 

### ISSUE THREAD

**Issue by XIAY0009** [stuarteiffert/RNN-for-Human-Activity-Recognition-using-2D-Pose-Input#19]
Dear Stuarteiffert, 

Thanks so much for sharing the codes on how to train the network, it has helped me understand a lot. I have one question which I hope to consult you on: 

**1) Training using the same dataset, I have drastic different results each time I run. Please see below diagrams. One of them was just up to 60% accuracy and the other run was 90% accuracy. May I know why is that so?**

![Differences](https://user-images.githubusercontent.com/36322922/54865348-2eb0fa80-4d9f-11e9-86e1-6debd3aeca9c.png)

![Training Session's Accuracy Over Iterations3](https://user-images.githubusercontent.com/36322922/54865321-bfd3a180-4d9e-11e9-8906-575317a576f1.png)

Thanks a lot 

Best, 
Calista

**Comment by stuarteiffert** [comment]
Hi Calista,

This looks like a case of instability in the gradients during training. This can sometimes occur in RNNs, even with LSTMs, usually a result of 'exploding gradients' (https://stats.stackexchange.com/questions/320919/why-can-rnns-with-lstm-units-also-suffer-from-exploding-gradients).

 This may may differ between training runs due to the random initialisation of weights at the start.
I haven't optimised this network or training (sorry!) but you should be able to minimise this by choosing different values of batch sizes, learning rates (and even changing the optimiser) and applying gradient clipping during training.

**Comment by XIAY0009** [comment]
Thanks Stuarteiffert for your explanation :)
